music_assistant/server/providers/plex/__init__.py

```diff
@@ -704,47 +758,54 @@ async def search(
         :param media_types: A list of media_types to include.
         :param limit: Number of items to return in the search (per type).
         """
-        tasks = {}
+        artists = None
+        albums = None
+        tracks = None
+        playlists = None
 
         async with TaskGroup() as tg:
-            for media_type in media_types:
-                if media_type == MediaType.ARTIST:
-                    tasks[MediaType.ARTIST] = tg.create_task(
-                        self._search_and_parse(
-                            self._search_artist(search_query, limit), self._parse_artist
-                        )
+            if MediaType.ARTIST in media_types:
+                artists = tg.create_task(
+                    self._search_and_parse(
+                        self._search_artist(search_query, limit), self._parse_artist
                     )
-                elif media_type == MediaType.ALBUM:
-                    tasks[MediaType.ARTIST] = tg.create_task(
```

Comment:
Think this found a legit bug - all search results were getting placed in tasks[MediaType.ARTIST].

Reply:
I'm also wondering why this is done in a taskgroup and not with asyncio gather but let's leave that for another rainy day